pkg/subscription/subscription.go

```diff
@@ -49,3 +54,48 @@ type RHSMConfig struct {
 	YumPlugins *SubManDNFPluginsConfig
 	SubMan     *SubManConfig
 }
+
+// RHSMConfigFromBP creates a RHSMConfig from a blueprint RHSMCustomization
+func RHSMConfigFromBP(bpRHSM *blueprint.RHSMCustomization) *RHSMConfig {
+	if bpRHSM == nil || bpRHSM.Config == nil {
+		return nil
+	}
+
+	config := &RHSMConfig{}
+
+	if plugins := bpRHSM.Config.DNFPlugins; plugins != nil {
+		config.DnfPlugins = &SubManDNFPluginsConfig{}
+		if plugins.ProductID != nil {
+			config.DnfPlugins.ProductID = &DNFPluginConfig{}
+			if plugins.ProductID.Enabled != nil {
+				config.DnfPlugins.ProductID.Enabled = common.ToPtr(*plugins.ProductID.Enabled)
+			}
+		}
+		if plugins.SubscriptionManager != nil {
+			config.DnfPlugins.SubscriptionManager = &DNFPluginConfig{}
+			if plugins.SubscriptionManager.Enabled != nil {
+				config.DnfPlugins.SubscriptionManager.Enabled = common.ToPtr(*plugins.SubscriptionManager.Enabled)
+			}
+		}
+	}
+
+	// NB: YUMPlugins are not exposed to end users as a customization
```

Comment:
Thanks for the comment, I was going to ask about this.

Reply:
I'm so sorry
<details>
  <summary>I'm so sorry</summary>
yump lugins
</details>